File: backend/internal/model/transaction.go
package model

import (
	"time"

	"github.com/google/uuid"
	"github.com/shopspring/decimal"
	"gorm.io/gorm"
)

type TransactionType string

const (
	Income   TransactionType = "INCOME"
	Expense  TransactionType = "EXPENSE"
	Transfer TransactionType = "TRANSFER"
)

type Transaction struct {
	ID              uuid.UUID       `gorm:"type:uuid;primary_key;" json:"id"`
	UserID          uuid.UUID       `gorm:"type:uuid;not null;index" json:"user_id"`
	AccountID       uuid.UUID       `gorm:"type:uuid;not null;index" json:"account_id"`
	TargetAccountID *uuid.UUID      `gorm:"type:uuid;index" json:"target_account_id,omitempty"` // For transfers
	Amount          decimal.Decimal `gorm:"type:decimal(15,2);not null" json:"amount"`
	Type            TransactionType `gorm:"type:varchar(20);not null" json:"type"`
	Category        string          `gorm:"type:varchar(100)" json:"category"`
	Note            string          `gorm:"type:text" json:"note"`
	TransactionDate time.Time       `gorm:"not null" json:"transaction_date"`
	CreatedAt       time.Time       `json:"created_at"`
	UpdatedAt       time.Time       `json:"updated_at"`
	DeletedAt       gorm.DeletedAt  `gorm:"index" json:"-"`
}

func (t *Transaction) BeforeCreate(tx *gorm.DB) (err error) {
	if t.ID == uuid.Nil {
		t.ID = uuid.New()
	}
	if t.TransactionDate.IsZero() {
		t.TransactionDate = time.Now()
	}
	return
}

File: frontend/src/types/transaction.ts
export type TransactionType = 'TRANSFER' | 'EXPENSE' | 'INCOME';

export interface Transaction {
  id: string;
  user_id: string;
  account_id: string;
  target_account_id?: string;
  amount: string;
  type: TransactionType;
  category: string;
  note: string;
  transaction_date: string;
  created_at: string;
  updated_at: string;
}

export interface Account {
  id: string;
  user_id: string;
  name: string;
  type: string; // 'NEC' | 'PLY' | 'FFA' | 'LTS' | 'EDU' | 'GIV'
  balance: string;
  created_at: string;
}
